src/v/cluster/rm_partition_frontend.cc

```diff
@@ -253,162 +253,20 @@ ss::future<begin_tx_reply> rm_partition_frontend::do_begin_tx(
 }
 
 ss::future<prepare_tx_reply> rm_partition_frontend::prepare_tx(
```

Comment:
`tx_gateway::prepare_tx` is the entry point of the request processing so it's better to put this exception there

Reply:
agree, done.